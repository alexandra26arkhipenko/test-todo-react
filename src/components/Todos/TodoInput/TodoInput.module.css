.todo-input {
  margin: auto;
  margin-top: 0px;
  width: 90%;
  max-width: 40rem;
}

.todo-input form {
  display: flex;
  align-items: center;
  padding: 37px;
}
.todo-input input[type="text"] {
  border: none;
  width: 100%;
  margin-left: 10px;
  font-size: 15px;
}

.todo-input input[type="text"]:focus {
  outline: none;
  color: rgb(81, 35, 3);
}

.todo-input ::placeholder {
  font-family: monospace;
}
