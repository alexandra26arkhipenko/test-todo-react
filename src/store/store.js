import { configureStore } from "@reduxjs/toolkit";
import { localStorageMiddleware } from "./middleware/localStorageMiddleware";

import todoReducer from "./todo";

const store = configureStore({
  reducer: todoReducer,
  middleware: [localStorageMiddleware],
});

export default store;
