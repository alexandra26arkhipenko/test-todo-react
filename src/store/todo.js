import { createSlice } from "@reduxjs/toolkit";
import { v4 as uuid } from "uuid";
import { getItemFromLocalStorage } from "../repository/localStorageRepository";
import { TODO_FOOTER_FILTER_ALL } from "../constants/constants";

const savedTodos = getItemFromLocalStorage("todos");
let todos = [];

if (savedTodos) {
  todos = JSON.parse(savedTodos);
}

const initialState = {
  todos: todos,
  filter: "",
};

const todoSlice = createSlice({
  name: "counter",
  initialState,
  reducers: {
    addTodo(state, action) {
      const newTodo = {
        id: uuid(),
        text: action.payload,
        isCompleted: false,
      };

      state.todos.unshift(newTodo);
    },
    compliteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos[index].isCompleted = !state.todos[index].isCompleted;
      }
    },
    deleteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos.splice(index, 1);
      }
    },
    deleteAllCompletedTodos(state) {
      state.todos = state.todos.filter((todo) => !todo.isCompleted);
    },
    filterTodos(state, action) {
      state.filter = action.payload;
    },
    reorderTodos(state, action) {
      let { dragIndex, dropIndex, filteredTodos } = action.payload;

      if (state.filterValue !== TODO_FOOTER_FILTER_ALL) {
        dragIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dragIndex].id
        );
        dropIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dropIndex].id
        );
      }

      const updatedTodos = [...state.todos];
      const [reorderedItem] = updatedTodos.splice(dragIndex, 1);
      updatedTodos.splice(dropIndex, 0, reorderedItem);
      state.todos = updatedTodos;
    },
  },
});

export const todoActions = todoSlice.actions;
export default todoSlice.reducer;
