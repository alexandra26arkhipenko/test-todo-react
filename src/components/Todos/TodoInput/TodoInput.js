import React, { useState } from "react";
import { useDispatch } from "react-redux";

import Card from "../../UI/Card/Card";
import { todoActions } from "../../../store/todo";
import classes from "./TodoInput.module.css";
import Checkbox from "../../UI/Checkbox/Checkbox";
import { TODO_INPUT_PLACEHOLDER } from "../../../constants/constants";

const TodoInput = () => {
  const dispatch = useDispatch();
  const [enteredTodo, setEnteredTodo] = useState("");

  const onSubmitHandler = (event) => {
    event.preventDefault();

    if (enteredTodo.trim().length === 0) {
      return;
    }

    dispatch(todoActions.addTodo(enteredTodo));
    setEnteredTodo('');
  };

  const inputHandler = (event) => {
    setEnteredTodo(event.target.value);
  };

  return (
    <Card className={classes["todo-input"]}>
      <form aria-label="form" onSubmit={onSubmitHandler}>
        <Checkbox isDisabled={true} checked={true}/>
        <input
          name="todo-input"
          placeholder={TODO_INPUT_PLACEHOLDER}
          className={classes["text-input"]}
          type="text"
          onChange={inputHandler}
          value={enteredTodo}
        ></input>
      </form>
    </Card>
  );
};

export default TodoInput;
