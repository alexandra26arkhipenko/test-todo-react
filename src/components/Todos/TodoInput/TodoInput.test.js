import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../store/store";
import TodoInput from "./TodoInput";

describe("TodoInput", () => {
  it("should render correctly", () => {
    const { container } = render(
        <Provider store={store}>
          <TodoInput />
        </Provider>
      );
    expect(container).toMatchSnapshot();
  });
});
