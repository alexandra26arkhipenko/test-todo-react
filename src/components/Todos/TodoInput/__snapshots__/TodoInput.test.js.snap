// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput should render correctly 1`] = `
<div>
  <div
    class="card todo-input"
  >
    <form>
      <input
        checked=""
        class="round"
        disabled=""
        type="checkbox"
      />
      <input
        class="text-input"
        placeholder="What are you going to do next?"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
